operator/src/main/java/oracle/kubernetes/weblogic/domain/api/WeblogicGenericApi.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2017, 2023, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.weblogic.domain.api;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.util.generic.GenericKubernetesApi;
+import io.kubernetes.client.util.generic.KubernetesApiResponse;
+import oracle.kubernetes.operator.helpers.HeaderModifierInterceptor;
+import oracle.kubernetes.weblogic.domain.model.KubernetesListObjectImpl;
+import oracle.kubernetes.weblogic.domain.model.PartialObjectMetadata;
+
+public class WeblogicGenericApi extends GenericKubernetesApi<PartialObjectMetadata, KubernetesListObjectImpl> {
+  private final ApiClient apiClient;
+
+  /**
+   * Constructs a generic api client used to get the partial CRD metadata.
+   * @param apiClient api client.
+   */
+  public WeblogicGenericApi(ApiClient apiClient) {
+    super(PartialObjectMetadata.class, KubernetesListObjectImpl.class, "apiextensions.k8s.io", "v1",
+        "customresourcedefinitions", apiClient);
+    this.apiClient = apiClient;
+  }
+
+  /**
+   * Get CRD metadata.
+   *
+   * @return CRD metadata
+   */
+  public PartialObjectMetadata readCustomResourceDefinitionMetadata(String name) {
+    HeaderModifierInterceptor.setPartialMetadataHeader(true);
+    final PartialObjectMetadata partialObjectMetadata = toPartialObjectMetadata(get(name));
+    HeaderModifierInterceptor.removePartialMetadataHeader();
```

Comment:
Suggest putting the remove call in a finally block so that you can't miss removing the interceptor if the call fails.

Reply:
Changed in [2e419a7](https://github.com/oracle/weblogic-kubernetes-operator/pull/3870/commits/2e419a7998d607aaa4296f5f6e7f9a885af8c8d0). Thanks.